### ISSUE TITLE
Generating adversarial examples for av_hubert 

### ISSUE THREAD

**Issue by ashwath98** [facebookresearch/av_hubert#56]
Hi, I am interested in generating adversarial examples for av hubert as I believe it will be more robust than other methods because of how it has been trained.

Because of my unfamiliarity with the fairseq framework, it's not completely clear to me where the model gets input during prediction (like in the colab notebook). Could you point me to the relevant part of the code base that defines the task?

-Ashwath

**Comment by chevalierNoir** [comment]
Hi,

The data loading is defined [here](https://github.com/facebookresearch/av_hubert/blob/main/avhubert/hubert_dataset.py#L348-L356). Here are the model forward functions during [pre-training](https://github.com/facebookresearch/av_hubert/blob/main/avhubert/hubert.py#L591-L599) and [fine-tuning](https://github.com/facebookresearch/av_hubert/blob/main/avhubert/hubert_asr.py#L493-L498).

**Comment by ashwath98** [comment]
Hey, thanks I was able to generate adversarial examples, surprisingly only a 0.01 epsilon led to a completely wrong sentence for the collab test video (I have attached an image of the sample difference between frames). I am trying out adversarial training and generating some visualizations for what the network is looking at to investigate further. If you have any suggestions for tools, I can use for visualizing which pixels the network is paying importance to, that will be immensely helpful.

thanks
-Ashwath


![original](https://user-images.githubusercontent.com/26197766/181354604-36ec7b13-175f-4372-a749-7ef65272e481.png)
![adversarial_sample](https://user-images.githubusercontent.com/26197766/181354614-d104e329-1ff5-46e3-8611-c478105fa0c8.png)
h
